.github/workflows/scripts/gcp-get-cached-disks.sh

```diff
@@ -34,31 +34,35 @@ find_cached_disk_image() {
         echo "Found ${git_source} Disk: ${disk_name}" >&2
         disk_description=$(gcloud compute images describe "${disk_name}" --format="value(DESCRIPTION)")
         echo "Description: ${disk_description}" >&2
-        echo "${disk_name}"  # This is the actual return value when a disk is found
+        echo "${disk_name}" # This is the actual return value when a disk is found
     else
         echo "No ${git_source} disk found with '${disk_search_pattern}' pattern." >&2
     fi
 }
 
-# Check if both $DISK_PREFIX and $DISK_SUFFIX are set, as they are required to find a cached disk image
+# Check if both $DISK_PREFIX and $DISK_SUFFIX are set, as they are required to
+# find a cached disk image.
 if [[ -n "${DISK_PREFIX}" && -n "${DISK_SUFFIX}" ]]; then
     # Find the most suitable cached disk image
     echo "Finding a ${DISK_PREFIX}-${DISK_SUFFIX} disk image for ${NETWORK}..."
     CACHED_DISK_NAME=""
 
-    # Check if main branch images are preferred
+    # Try to find an image based on the `main` branch if that branch is preferred.
     if [[ "${PREFER_MAIN_CACHED_STATE}" == "true" ]]; then
         CACHED_DISK_NAME=$(find_cached_disk_image "main-[0-9a-f]+" "main branch")
-    # Else, try to find a cached disk image from the current branch (or PR)
-    else
+    fi
+
+    # If no image was found, try to find one from the current branch (or PR).
+    if [[ -z "${CACHED_DISK_NAME}" ]]; then
```

Comment:
To avoid repeating the if condition several times, we can also do:
```shell
# Try to find an image based on the `main` branch if that branch is preferred.
if [[ "${PREFER_MAIN_CACHED_STATE}" == "true" ]]; then
    CACHED_DISK_NAME=$(find_cached_disk_image "main-[0-9a-f]+" "main branch")
fi
# If no image was found, try to find one from the current branch (or PR).
CACHED_DISK_NAME=${CACHED_DISK_NAME:-$(find_cached_disk_image ".+-${GITHUB_REF}" "branch")}
# If we still have no image, try to find one from any branch.
CACHED_DISK_NAME=${CACHED_DISK_NAME:-$(find_cached_disk_image ".+-[0-9a-f]+" "any branch")}

# Handle the case where no suitable disk image is found
if [[ -z "${CACHED_DISK_NAME}" ]]; then
    echo "No suitable cached state disk available. Try running the cached state rebuild job."
    exit 1
else
    echo "Selected Disk: ${CACHED_DISK_NAME}"
fi
```

Here we'd use `${VAR:-value}` to set `CACHED_DISK_NAME` to the result of the next `find_cached_disk_image` call only if it's not already set.